Automattic/wp-calypso
# Style selection in design picker: Implement style variation in the full-screen theme preview

#### Proposed Changes

**NOTE: This PR depends on D86697-code**

This PR set the groundwork for the new full-screen theme preview, which can be referenced [here](7l53h5fxAUNjVRBQ82YFwn-fi-248%3A40987). In the spirit of keeping this PR reasonably scoped, we are only implementing the "sidebar" of the component, which contains the theme title, description, style variations, support links, and theme features.

While the component "sidebar" is mostly complete, there are some work left that will be addressed in follow-up PRs:
* Style variation preview font rendering.
* Selecting style variation from the design picker should carry over to the full-screen theme preview.

---

This PR main does 3 things:
1. Creates a new package `design-preview`, which is the full-screen theme preview users will see when they click on a theme from the theme grid. The relevant files to the package are:
![Screen Shot 2022-09-01 at 9 41 18 PM](https://user-images.githubusercontent.com/797888/187928781-766bba7d-0bbf-460f-b670-71e98396a622.png)

2. Retrieves theme details from the `/v2/starter-designs/${slug}` endpoint (diff: D86697-code). The relevant files are:
![Screen Shot 2022-09-01 at 9 43 27 PM](https://user-images.githubusercontent.com/797888/187929247-0dc0f6e0-3149-4bb0-bdc5-f667f69f60dc.png)

3. Refactor the style variation badges implemented in #66856 now that we have a better understanding of the variation data structure. The relevant files are: 
![Screen Shot 2022-09-01 at 9 46 19 PM](https://user-images.githubusercontent.com/797888/187929882-71291398-5d20-4e95-aeb1-9695c51586ea.png)

#### Testing Instructions



* Head to the unified design picker `/setup/designSetup?siteSlug=${site_slug}`
* Ensure that clicking any generated design should display the old full-screen theme preview.
* Ensure that clicking any static design should display the new full-screen theme preview.
* Ensure that themes with style variation indicators should also display style variation iframe previews in the new full-screen theme preview.
  
#### Pre-merge Checklist






Related to #66741
